pkg/codegen/testing/test/testdata/external-resource-schema/go/example/workload.go

```diff
@@ -149,8 +149,8 @@ func (o WorkloadOutput) ToWorkloadOutputWithContext(ctx context.Context) Workloa
 	return o
 }
 
-func (o WorkloadOutput) Pod() corev1.PodPtrOutput {
-	return o.ApplyT(func(v *Workload) corev1.PodPtrOutput { return v.Pod }).(corev1.PodPtrOutput)
+func (o WorkloadOutput) Pod() corev1.PodTypePtrOutput {
```

Comment:
I don't see it, but are we now outputting this as a type on sdkgen with this PR? I don't see `PodTypePtrOutput` in the exported types of `kubernetes/core/v1`:

https://pkg.go.dev/github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1

Reply:
Yeah, that type doesn't exist. Neither `corev1.PodPtrOutput` or `corev1.PodTypePtrOutput` exist (this test doesn't compile). The appending of type is correct. Kubernetes `core/v1` has both a `Pod` defined in both [resources](https://github.com/pulumi/pulumi-kubernetes/blob/7547fb6ae397a8592c20ea9b5ddbdaa8605ecc33/provider/cmd/pulumi-resource-kubernetes/schema.json#L38531) and [types](https://github.com/pulumi/pulumi-kubernetes/blob/7547fb6ae397a8592c20ea9b5ddbdaa8605ecc33/provider/cmd/pulumi-resource-kubernetes/schema.json#L16662), and workload references the type:
https://github.com/pulumi/pulumi/blob/0e64874da720de95c2ae02761d8aced498765cee/pkg/codegen/testing/test/testdata/external-resource-schema/schema.json#L62-L69

We are correctly appending the type because we are generating the external "Pod" type with the knowledge that another "Pod" from resources exists.

Reply:
It does point out that this fix won't operate correctly for output types (which is why we are not generating `*corev1.PodTypeOutput`). I'll add that fix.